scala/lambdas/ingest-flow-control/src/test/scala/uk/gov/nationalarchives/ingestflowcontrol/LambdaTest.scala

```diff
@@ -0,0 +1,425 @@
+package uk.gov.nationalarchives.ingestflowcontrol
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers.*
+import cats.effect.*
+import cats.effect.unsafe.implicits.global
+import io.circe.{Decoder, Encoder}
+import org.scalatest.EitherValues
+import org.scanamo.DynamoFormat
+import org.scanamo.request.RequestCondition
+import software.amazon.awssdk.services.dynamodb.model.BatchWriteItemResponse
+import software.amazon.awssdk.services.sfn.model.StartExecutionResponse
+import uk.gov.nationalarchives.{DADynamoDBClient, DASFNClient, DASSMClient}
+import uk.gov.nationalarchives.ingestflowcontrol.Lambda.*
+import uk.gov.nationalarchives.dynamoformatters.DynamoFormatters.*
+import uk.gov.nationalarchives.ingestflowcontrol.helpers.StepFunctionExecution
+
+import java.time.{Duration, Instant}
+
+class LambdaTest extends AnyFlatSpec with EitherValues:
+
+  val config: Config = Config("", "", "")
+
+  def notImplemented[T]: IO[Nothing] = IO.raiseError(new Exception("Not implemented"))
+
+  def predictableRandomNumberSelector(selected: Int = 10): (Int, Int) => Int = { (ignoredMin, ignoredMax) => if selected > ignoredMax then ignoredMax else selected }
+
+  private def runLambda(
+      input: Input,
+      rowsInTable: List[IngestQueueTableItem],
+      initialConfig: FlowControlConfig,
+      initialExecutions: List[StepFunctionExecution],
+      randomSelection: (Int, Int) => Int,
+      errors: Option[Errors] = None
+  ): (Either[Throwable, Unit], List[IngestQueueTableItem], FlowControlConfig, List[StepFunctionExecution]) = {
```

Comment:
Maybe a case class called "Dependencies" or something?

Reply:
Not sure I understand this, can you please explain more.

Reply:
Because you are returning a tuple, so it might be nicer/clearer to just put all of these in a case class so at least it's named and later on in the code, instead of `tuple._1`, `tuple._2`, you can do `dependencies.error`, `dependencies.items` etc

Reply:
Do you mean `LambdaRunResults` class ?  (line 48)

Reply:
I think I understand, you mean use the class type itself as a return value from that method?

Reply:
Yes. Sorry for not being clear. I meant instead of returning a tuple, you could return a case class called "Dependencies" or something

Reply:
...And since this method called by `LambdaRunResults`, then that could take the class as an argument rather than a tuple

Reply:
I just realised that there is another case class already called "Dependencies"; maybe `Results`?